source/common/http/filter/lua/wrappers.h

```diff
@@ -98,6 +100,63 @@ class HeaderMapWrapper : public Envoy::Lua::BaseLuaObject<HeaderMapWrapper> {
   friend class HeaderMapIterator;
 };
 
+/**
+ * Iterator over a metadata map.
+ */
+class MetadataMapIterator : public Envoy::Lua::BaseLuaObject<MetadataMapIterator> {
```

Comment:
I would move the metadata wrappers here: https://github.com/envoyproxy/envoy/blob/master/source/common/lua/wrappers.h

I don't think they are HTTP specific